homeassistant/components/api/__init__.py

```diff
@@ -387,6 +387,25 @@ async def post(
             )
 
         context = self.context(request)
+
+        if response_requested := "return_response" in request.query:
+            if (
+                hass.services.supports_response(domain, service)
+                is ha.SupportsResponse.NONE
+            ):
+                return self.json_message(
+                    "Service does not support responses. Remove return_response from request.",
+                    HTTPStatus.BAD_REQUEST,
+                )
+        elif (
+            hass.services.supports_response(domain, service) == ha.SupportsResponse.ONLY
```

Comment:
```suggestion
            hass.services.supports_response(domain, service) is ha.SupportsResponse.ONLY
```

both sides are an enum